test/TestDataGenerator.cpp

```diff
@@ -0,0 +1,444 @@
+#include "../src/DataGenerator.h"
+#include "../src/utils.h"
+#include "gtest/gtest.h"
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <tuple>
+#include <utility>
+
+TEST(TestDataGenerator, TestDummy) {
+  std::ofstream file;
+  file.open("../data/test.txt");
+  file
+      << "Please writr this text to a file.\n this text is written using C++\n";
+  file.close();
+  bool u = fileExists("../data/test.txt");
+  std::ostringstream os2;
+  os2 << u;
+  EXPECT_EQ(os2.str(), "1");
+}
+
+/**
+TEST(TestDataGenerator, TestDatasetGenerator) {
```

Comment:
Let's just remove or keep test, let's not add commented tests in the codebase

Reply:
This is used to generate set, id like to keep it for now until random number generator is done